internal/server/iigointernal/judiciary.go

```diff
@@ -0,0 +1,196 @@
+	return x
+}
+
+// withdrawPresidentSalary withdraws the president's salary from the common pool.
+func (j *judiciary) withdrawPresidentSalary(gameState *gamestate.GameState) error {
+	var presidentSalary = shared.Resources(rules.VariableMap["presidentSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(presidentSalary, gameState)
+	if withdrawError == nil {
+		j.presidentSalary = presidentSalary
+	}
+	return withdrawError
+}
+
+// sendPresidentSalary sends the president's salary to the president.
+func (j *judiciary) sendPresidentSalary() {
+	if j.clientJudge != nil {
+		amount := j.clientJudge.PayPresident()
+		executiveBranch.budget = amount
+		return
+	}
+	amount, _ := j.PayPresident()
+	executiveBranch.budget = amount
+}
+
+// PayPresident pays the president salary.
+func (j *judiciary) PayPresident() (shared.Resources, error) {
+	hold := j.presidentSalary
+	j.presidentSalary = 0
+	return hold, nil
+}
+
+// setSpeakerAndPresidentIDs set the speaker and president IDs.
+func (j *judiciary) setSpeakerAndPresidentIDs(speakerId shared.ClientID, presidentId shared.ClientID) {
+	j.speakerID = speakerId
+	j.presidentID = presidentId
+}
+
+// InspectHistory checks all actions that happened in the last turn and audits them.
+// This can be overridden by clients.
+func (j *judiciary) inspectHistory() (map[shared.ClientID]roles.EvaluationReturn, error) {
+	j.budget -= serviceCharge
+	return j.clientJudge.InspectHistory()
+}
+
+// inspectBallot checks each ballot action adheres to the rules
+func (j *judiciary) inspectBallot() (bool, error) {
+	// 1. Evaluate difference between newRules and oldRules to check
+	//    rule changes are in line with RuleToVote in previous ballot
+	// 2. Compare each ballot action adheres to rules in ruleSet matrix
+	j.budget -= serviceCharge // will be removed post-MVP
+	rulesAffectedBySpeaker := j.EvaluationResults[j.speakerID]
+	indexOfBallotRule, err := searchForRule("inspect_ballot_rule", rulesAffectedBySpeaker.Rules)
+	if err == nil {
+		return rulesAffectedBySpeaker.Evaluations[indexOfBallotRule], nil
+	} else {
+		return true, errors.Errorf("Speaker did not conduct any ballots")
+	}
+}
+
+// inspectAllocation checks each resource allocation action adheres to the rules
+func (j *judiciary) inspectAllocation() (bool, error) {
+	// 1. Evaluate difference between commonPoolNew and commonPoolOld
+	//    to check resource allocation changes are in line with ResourceRequests
+	//    in previous resourceAllocation
+	// 2. Compare each resource allocation action adheres to rules in ruleSet
+	//    matrix
+	j.budget -= serviceCharge // will be removed post-MVP
+	rulesAffectedByPresident := j.EvaluationResults[j.presidentID]
+	indexOfAllocRule, err := searchForRule("inspect_allocation_rule", rulesAffectedByPresident.Rules)
+	if err == nil {
+		return rulesAffectedByPresident.Evaluations[indexOfAllocRule], nil
+	} else {
+		return true, errors.Errorf("President didn't conduct any allocations")
+	}
```

Comment:
You should do it this way since it follows Go style guide.
```suggestion
	if err != nil {
		return true, errors.Errorf("President didn't conduct any allocations")
	} 
	return rulesAffectedByPresident.Evaluations[indexOfAllocRule], nil
```